feature/experimental/bgp/ate_tests/bgp_tcp_mss_path_mtu/bgp_tcp_mss_path_mtu_test.go

```diff
@@ -0,0 +1,406 @@
+	vlan20             = 20
+	ISISInstance       = "DEFAULT"
+	authPWd            = "BGPTCPMSS"
+	dut1AreaAddress    = "49.0001"
+	dut1SysID          = "1920.0000.2001"
+	dut2AreaAddress    = "49.0001"
+	dut2SysID          = "1920.0000.3001"
+)
+
+var (
+	dut1Port1 = attrs.Attributes{
+		Desc:    "DUT to ATE source",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::192:0:2:1",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	atePort1 = attrs.Attributes{
+		Name:    "ateSrc",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::192:0:2:2",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	dut1Port2 = attrs.Attributes{
+		Desc:    "DUT1 to DUT2",
+		IPv4:    "192.0.2.5",
+		IPv4Len: plenIPv4,
+	}
+	dut2Port1 = attrs.Attributes{
+		Name:    "DUT2 to DUT1",
+		IPv4:    "192.0.2.6",
+		IPv4Len: plenIPv4,
+	}
+)
+
+// configureDUT configures all the interfaces on the DUT.
+func configureDUT(t *testing.T) {
+	dc := gnmi.OC()
+	dut1 := ondatra.DUT(t, "dut1")
+	dut2 := ondatra.DUT(t, "dut2")
+
+	t.Log("Configure interfaces on dut1.")
+	i1 := dut1Port1.NewOCInterface(dut1.Port(t, "port1").Name(), dut1)
+	gnmi.Replace(t, dut1, dc.Interface(i1.GetName()).Config(), i1)
+	i2 := dut1Port2.NewOCInterface(dut1.Port(t, "port2").Name(), dut1)
+	gnmi.Replace(t, dut1, dc.Interface(i2.GetName()).Config(), i2)
+
+	t.Log("Configure interfaces on dut2.")
+	i3 := dut2Port1.NewOCInterface(dut2.Port(t, "port1").Name(), dut2)
+	gnmi.Replace(t, dut2, dc.Interface(i3.GetName()).Config(), i3)
+}
+
+// bgpCreateNbr creates bgp configuration on dut device.
+func bgpCreateNbr(t *testing.T, dut *ondatra.DUTDevice, authPwd, routerId string, localAs uint32, nbrs []*bgpNeighbor) *oc.NetworkInstance_Protocol {
+
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+	global := bgp.GetOrCreateGlobal()
+	global.RouterId = ygot.String(routerId)
+	global.As = ygot.Uint32(localAs)
+	global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+
+	// Note: we have to define the peer group even if we aren't setting any policy because it's
+	// invalid OC for the neighbor to be part of a peer group that doesn't exist.
+	for _, nbr := range nbrs {
+		if nbr.isV4 {
+			// Create Peer group.
+			pg1 := bgp.GetOrCreatePeerGroup(nbr.peerGrp)
+			pg1.PeerAs = ygot.Uint32(nbr.as)
+			pg1.PeerGroupName = ygot.String(nbr.peerGrp)
+			// configure neighbor.
+			nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
+			nv4.PeerGroup = ygot.String(nbr.peerGrp)
+			nv4.PeerAs = ygot.Uint32(nbr.as)
+			nv4.Enabled = ygot.Bool(true)
+			if authPwd != "" {
+				nv4.AuthPassword = ygot.String(authPWd)
+			}
+			af4 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+			af4.Enabled = ygot.Bool(true)
+			af6 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+			af6.Enabled = ygot.Bool(false)
+		} else {
```

Comment:
why doesn't the peer group need to be created in the v6 case?

Reply:
its needed, updated it.